python-sdk/tests/e2e/test_async_experiment_adk_e2e.py

```
@@ -99,22 +99,23 @@ def _poll_run_results(
-                rows = last_body.get("dataset", [])
-                if len(rows) >= expected_items:
-                    if not require_cost or all(row.get("cost") for row in rows):
-                        return last_body
+                parsed = response.json()
+                if isinstance(parsed, dict):
+                    last_body = parsed
+                    rows = last_body.get("dataset", [])
+                    if len(rows) >= expected_items:
+                        if not require_cost or all(row.get("cost") for row in rows):
+                            return last_body
         except Exception as exc:
```

Ruff (0.15.12)

[warning] Do not catch blind exception: `Exception`

(BLE001)